conways-game-of-life/src/scripts/grid.js

```diff
@@ -0,0 +1,95 @@
+const gridLife = gridArray => {
+  const grid = gridArray;
+  const width = gridArray.length;
+  const height = gridArray[0].length;
+
+  const validateCoords = (x, y) => {
+    let xValid = x;
+    let yValid = y;
+    // left or Right
+    while (xValid < 0) {
+      xValid += width;
+    }
+    while (xValid >= width) {
+      xValid -= width;
+    }
+
+    // top or Bottom
+    while (yValid < 0) {
+      yValid += height;
+    }
+    while (yValid >= height) {
+      yValid -= height;
+    }
+    return { xValid, yValid };
+  };
+
+  const getCellValue = (x, y) => {
+    const { xValid, yValid } = validateCoords(x, y);
+    // console.log(`x: ${xValid}, y: ${yValid}, grid: ${grid}`)
+    return grid[xValid][yValid];
+  };
+
+  const getSurroundingValues = (x, y) => {
+    return [
+      getCellValue(x - 1, y - 1),
+      getCellValue(x - 1, y),
+      getCellValue(x - 1, y + 1),
+      getCellValue(x, y - 1),
+      getCellValue(x, y + 1),
+      getCellValue(x + 1, y - 1),
+      getCellValue(x + 1, y),
+      getCellValue(x + 1, y + 1),
+    ];
+  };
+  const countSurroundingLives = (x, y) => {
+    const values = getSurroundingValues(x, y);
+    let counter = 0;
+    for (let i = 0; i < values.length; i += 1) {
```

Comment:
Just a small suggestion, but using "i++" instead of "i + 1" seems to be the convention with these types of loops

Reply:
yeah, I definitely prefered to use i++ but on the course that I have just finished the linting rules that they used include to use i += 1, it's just a matter of taste